src/Control/Monad/Freer/Trace.hs

```diff
@@ -28,7 +29,6 @@ trace :: Member Trace effs => String -> Eff effs ()
 trace = send . Trace
 
 -- | An 'IO' handler for 'Trace' effects.
-runTrace :: Eff '[Trace] a -> IO a
-runTrace (Val x) = return x
-runTrace (E u q) = case extract u of
-  Trace s -> putStrLn s >> runTrace (qApp q ())
+runTrace :: Member IO r => Eff (Trace ': r) ~> Eff r
+runTrace = natural @IO $ \(Trace s) -> putStrLn s
```

Comment:
If this is going to be in the std lib, there really should be a `(w -> String) -> Writer w ~> Trace` interpreter